lib/utils/src/graph/algorithms.cc

```diff
@@ -321,11 +373,23 @@ std::vector<DirectedEdge> get_edge_topological_ordering(DiGraphView const &g) {
     }
   }
 
+
   assert(result.size() == get_edges(g).size());
 
   return result;
 }
 
+std::vector<Node> get_neighbors(DiGraphView const & g, Node const & n) {
+  std::vector<Node> neighbors;
```

Comment:
Why not an `unordered_set`? Also,  `return set_union(get_outgoing_edges(g, n), get_incoming_edges(g, n));`

In general, I recommend you give `containers.h` a read, it can be used to shorten a bunch of this code

Reply:
Good point, then 
```cpp
return set_union(
  transform(get_outgoing_edges(g, n), [](Node const &n) { return n.dst; }),
  transform(get_incoming_edges(g, n), [](Node const &n) { return n.src; })
);
```